warp10/src/main/java/io/warp10/standalone/StandaloneStoreClient.java

```diff
@@ -392,44 +401,19 @@ public boolean hasNext() {
               return false;
             }
             
-            if (fromArchive) {
-              startrow = new byte[Constants.HBASE_RAW_DATA_KEY_PREFIX.length + 8 + 8];
-              ByteBuffer bb = ByteBuffer.wrap(startrow).order(ByteOrder.BIG_ENDIAN);
-              bb.put(Store.HBASE_ARCHIVE_DATA_KEY_PREFIX);
-              bb.putLong(metadatas.get(idx).getClassId());
-              bb.putLong(metadatas.get(idx).getLabelsId());
-              
-              stoprow = new byte[startrow.length + 8]; // Provision for 2**64 chunks (!)
-              bb = ByteBuffer.wrap(stoprow).order(ByteOrder.BIG_ENDIAN);
-              bb.put(Store.HBASE_ARCHIVE_DATA_KEY_PREFIX);
-              bb.putLong(metadatas.get(idx).getClassId());
-              bb.putLong(metadatas.get(idx).getLabelsId());              
-              bb.putLong(0xffffffffffffffffL);
-            } else {
-              startrow = new byte[Constants.HBASE_RAW_DATA_KEY_PREFIX.length + 8 + 8 + 8];
-              ByteBuffer bb = ByteBuffer.wrap(startrow).order(ByteOrder.BIG_ENDIAN);
-              bb.put(Constants.HBASE_RAW_DATA_KEY_PREFIX);
-              bb.putLong(metadatas.get(idx).getClassId());
-              bb.putLong(metadatas.get(idx).getLabelsId());
-              bb.putLong(Long.MAX_VALUE - now);
-              
-              stoprow = new byte[startrow.length];
-              bb = ByteBuffer.wrap(stoprow).order(ByteOrder.BIG_ENDIAN);
-              bb.put(Constants.HBASE_RAW_DATA_KEY_PREFIX);
-              bb.putLong(metadatas.get(idx).getClassId());
-              bb.putLong(metadatas.get(idx).getLabelsId());              
+            startrow = new byte[Constants.HBASE_RAW_DATA_KEY_PREFIX.length + 8 + 8 + 8];
+            ByteBuffer bb = ByteBuffer.wrap(startrow).order(ByteOrder.BIG_ENDIAN);
+            bb.put(Constants.HBASE_RAW_DATA_KEY_PREFIX);
+            bb.putLong(metadatas.get(idx).getClassId());
+            bb.putLong(metadatas.get(idx).getLabelsId());
+            bb.putLong(Long.MAX_VALUE - now);
               
-              //
-              // If 'timespan' is negative, we are interested in retrieving -'timespan' values,
-              // regardless of how long ago they were measured
-              //
-              
-              if (timespan >= 0) {
-                bb.putLong(Long.MAX_VALUE - (now - timespan + 1));
-              } else {
-                bb.putLong(0xffffffffffffffffL);
-              }
-            }            
+            stoprow = new byte[startrow.length];
+            bb = ByteBuffer.wrap(stoprow).order(ByteOrder.BIG_ENDIAN);
+            bb.put(Constants.HBASE_RAW_DATA_KEY_PREFIX);
+            bb.putLong(metadatas.get(idx).getClassId());
+            bb.putLong(metadatas.get(idx).getLabelsId());                            
+            bb.putLong(Long.MAX_VALUE - then);
```

Comment:
Either change the key or make sure `then` in included in the "core" fetch and not the pre-boundary one.

Reply:
The previous comparison with stoprow actually takes care of this.